framework/wazuh/core/cluster/common.py

```diff
@@ -1047,6 +1055,334 @@ def get_node(self):
         return self.get_manager().get_node()
 
 
+class SyncTask:
+    """
+    Common class for master/worker sync tasks.
+    """
+
+    def __init__(self, cmd: bytes, logger, manager):
+        """Class constructor.
+
+        Parameters
+        ----------
+        cmd : bytes
+            Request command to send to the master/worker.
+        logger : Logger object
+            Logger to use during synchronization process.
+        manager : MasterHandler/WorkerHandler object
+            The MasterHandler/WorkerHandler object that creates this one.
+        """
+        self.cmd = cmd
+        self.logger = logger
+        self.manager = manager
+
+    async def request_permission(self):
+        """Request permission to start synchronization process with the master.
+
+        Returns
+        -------
+        bool
+            Whether permission is granted.
+        """
+        result = await self.manager.send_request(command=self.cmd + b'_p', data=b'')
+
+        if isinstance(result, Exception):
+            self.logger.error(f"Error asking for permission: {result}")
+        elif result == b'True':
+            self.logger.debug("Permission to synchronize granted.")
+            return True
+        else:
+            self.logger.debug(f"Master didn't grant permission to start a new synchronization: {result}")
+
+        return False
```

Comment:
Right but that would imply modifying the worker.py file and its test, we want both issues to be able to work independently, so this will be solved in: https://github.com/wazuh/wazuh/issues/11792